### ISSUE TITLE
Hide column inside Stack doesn't work

### ISSUE THREAD

**Issue by Algeneralo** [laravel/nova-issues#3242]
- Laravel Version: 8.22
- Nova Version: 3.21.1
- PHP Version: 7.4
- Database Driver & Version: pgsql
- Operating System and Version: Windows 10
- Browser type and version: Opera

### Description:
Hiding a column inside `Stack` doesn't work, I have to move it outside the `Stack` to make it work

### Detailed steps to reproduce the issue on a fresh Nova installation:
Add the following code

```php
Stack::make('Specialist', [
    BelongsTo::make('Specialist')
    ,
  Boolean::make(__('Specific Specialist'), 'specific_specialist')
      ->hideFromIndex(function () {
          return true;
      })
    ,
  ])
```
Move this code outside the `stack`, then it will work exactly as expected

```php
Boolean::make(__('Specific Specialist'), 'specific_specialist')
      ->hideFromIndex(function () {
          return true;
      })
```

**Comment by Algeneralo** [comment]
@davidhemphill Could you please replay with closing reason, or if it is not an issue, just tell me with the solution?
you just closed it without any comment and I'm still waiting for a response ......

**Comment by crynobone** [comment]
@Algeneralo Closed Assigned Bug only one thing, which is bugfix has been merged to `laravel/nova`. 

In fact, this fix just been released with `v3.22.0` today.

![image](https://user-images.githubusercontent.com/172966/108166653-71b4ad80-712f-11eb-8b87-e521e9456685.png)

**Comment by Algeneralo** [comment]
@crynobone Thanks for explaining that!
